utils/keystore.py

```diff
@@ -1,167 +0,0 @@
-import json
-from dataclasses import (
-    asdict,
-    dataclass,
-    fields,
-    field as dataclass_field
-)
-from secrets import randbits
-from typing import Any, Dict, Union
-from unicodedata import normalize
-from uuid import uuid4
-
-from utils.crypto import (
-    AES_128_CTR,
-    PBKDF2,
-    scrypt,
-    SHA256,
-)
-
-UNICODE_CONTROL_CHARS = list(range(0x00, 0x20)) + list(range(0x7F, 0xA0))
-
-# https://github.com/ethereum/eth2.0-deposit-cli/blob/master/eth2deposit/key_handling/keystore.py
-# Commit: b6f7ca9
-
-hexdigits = set('0123456789abcdef')
-
-
-def encode_bytes(obj: Union[str, Dict[str, Any]]) -> Union[bytes, str, Dict[str, Any]]:
-    """
-    Recursively encodes objects that contain hexstrings into objects that contain bytes.
-    """
-    if isinstance(obj, str) and all(c in hexdigits for c in obj):
-        return bytes.fromhex(obj)
-    elif isinstance(obj, dict):
-        for key, value in obj.items():
-            obj[key] = encode_bytes(value)
-    return obj
-
-
-class BytesDataclass:
-    """
-    BytesDataClasses are DataClass objects that automatically encode hexstrings into bytes,
-    and have an `as_json` function that encodes bytes back into hexstrings.
-    """
-
-    def __post_init__(self) -> None:
-        for field in fields(self):
-            if field.type in (bytes, Dict[str, Any]):
-                # Convert hexstring to bytes
-                self.__setattr__(field.name, encode_bytes(self.__getattribute__(field.name)))
-
-    def as_json(self) -> str:
-        return json.dumps(asdict(self), default=lambda x: x.hex())
-
-
-@dataclass
-class KeystoreModule(BytesDataclass):
-    function: str = ''
-    params: Dict[str, Any] = dataclass_field(default_factory=dict)
-    message: bytes = bytes()
-
-
-@dataclass
-class KeystoreCrypto(BytesDataclass):
-    kdf: KeystoreModule = KeystoreModule()
-    checksum: KeystoreModule = KeystoreModule()
-    cipher: KeystoreModule = KeystoreModule()
-
-    @classmethod
-    def from_json(cls, json_dict: Dict[Any, Any]) -> 'KeystoreCrypto':
-        kdf = KeystoreModule(**json_dict['kdf'])
-        checksum = KeystoreModule(**json_dict['checksum'])
-        cipher = KeystoreModule(**json_dict['cipher'])
-        return cls(kdf=kdf, checksum=checksum, cipher=cipher)
-
-
-@dataclass
-class Keystore(BytesDataclass):
-    """
-    Implement an EIP 2335-compliant keystore. A keystore is a JSON file that
-    stores an encrypted version of a private key under a user-supplied password.
-
-    Ref: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2335.md
-    """
-    crypto: KeystoreCrypto = KeystoreCrypto()
-    uuid: str = ''
-    index: int = 0
```

Comment:
`index` and `threshold` are added here on top of the copied `Keystore`. I can make a separate PR which will clean up the copied functionality.

Reply:
OK, I gave you access to the forked repository (check email for the invite).

Reply:
Thanks, I've updated it in [fb3e7b8](https://github.com/stakewise/bls-horcrux/pull/4/commits/fb3e7b8c82bdbf73197ee605b65fe77e6d24d3b0)